### ISSUE TITLE
Input field "jumps" after doing Keyboard.dismiss()

### ISSUE THREAD

**Issue by ortutay** [FaridSafi/react-native-gifted-chat#1807]
#### Issue Description

The input field area "jumps" briefly while doing a Keyboard.dismiss().

This video shows the issue: https://orderzstatic.s3-us-west-2.amazonaws.com/misc/RPReplay_Final1590566027.MP4

The jump occurs around 4sec in. Downloading & slowly stepping through it makes it more obvious.

#### Steps to Reproduce / Code Snippets

I just did a Keyboard.dismiss() on a slightly modified version of the example here https://github.com/FaridSafi/react-native-gifted-chat/blob/master/App.tsx . In the video, the button in the red box triggers the dismiss. I'm running on iOS.

#### Expected Results

It should be a smooth animation

#### Additional Information

* Nodejs version: v12.16.3
* React version: 16.9.0
* React Native version: 0.61
* react-native-gifted-chat version: 0.16.1
* Platform(s) (iOS, Android, or both?): only tried iOS
* TypeScript version: N/A, not using typescript in my project

**Comment by ortutay** [comment]
I investigated this a little bit. It looks like an issue with the handling of keyboardWillShow and keyboardWillHide in GiftedChat.js and InputToolbar.js. If I comment out some lines it stops the jumping, though it also turns off the animation.

The lines I've commented out are:

- the setState in onKeyboardWillShow and onKeyboardWillHide in GiftedChat.js
- the setState in keyboardWillShow and keyboardWillHide in InputToolbar.js
